cmd/diff.go

```diff
@@ -102,3 +105,99 @@ files.`,
   # referred to by './kubeconfig'.
   ks diff --kubeconfig=./kubeconfig -f ./pod.yaml`,
 }
+
+func initDiffCmd(cmd *cobra.Command, wd metadata.AbsPath, envSpec *envSpec, diffStrategy string) (kubecfg.DiffCmd, error) {
+	const (
+		remote = "remote"
+		local  = "local"
+	)
+
+	var err error
+
+	// ---------------------------------------------------------------------------
+	// Diff between expanded Kubernete objects and objects on a remote cluster
+	// ---------------------------------------------------------------------------
+	if envSpec.env2 == nil {
+		c := kubecfg.DiffRemoteCmd{}
+		c.DiffStrategy = diffStrategy
+		c.Client = &kubecfg.Client{}
+
+		c.Client.APIObjects, err = expandEnvCmdObjs(cmd, envSpec, wd)
+		if err != nil {
+			return nil, err
+		}
+
+		c.Client.ClientPool, c.Client.Discovery, err = restClientPool(cmd, envSpec.env)
+		if err != nil {
+			return nil, err
+		}
+
+		c.Client.Namespace, err = namespace()
+		if err != nil {
+			return nil, err
+		}
+
+		return &c, nil
+	}
+
+	env1 := strings.SplitN(*envSpec.env, ":", 2)
```

Comment:
I think we might need to disallow the `:` character in environment identifiers, because of the example above, `ks diff local:default`, it gets confusing for users to reason about it. Here are the options I see:

1. Environment IDs can't begin with the prefix `local:`, which makes `ks diff local: default` invalid, but then we're in the weird position of disallowing a single, specific prefix in environment IDs, or
1. Environment IDs can have `:` in them and the prefix `local:` _is_ allowed, in which which `ks diff local:default` parses to look for the environment _called_ `local:default`, which is different from the _local version of the environment called `local:default`_ (which would be denoted as `local:local:default`).
1. Environment IDs can't have `:` in them, which is more restrictive, but less confusing in every case I can think of.

I don't know what you think, but I like the last option most.

Reply:
We already don't allow for punctuations in env names, which includes `:`.  I added a check to `ks diff local:env`